07.ls_object/list_command.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require_relative './file_info'
+
+class ListCommand
+  attr_reader :files
+
+  NUMBER_OF_COLUMNS = 3
+
+  DETAIL_KEYS = %i[
+    type_and_permission
+    hard_link_count
+    user_name
+    group_name
+    size
+    timestamp
+    name
+  ].freeze
+
+  private_constant :NUMBER_OF_COLUMNS, :DETAIL_KEYS
+
+  def initialize(file_names, path)
+    @files = file_names.map { |name| FileInfo.new(path, name) }
+  end
+
+  def format_file_names
+    column_spacing = files.map { |file| file.name.size }.max
+    number_of_rows = (files.size / NUMBER_OF_COLUMNS.to_f).ceil
+    spaced_files = files.map { |file| file.name.ljust(column_spacing) }
+    spaced_files.each_slice(number_of_rows).to_a
+                .each { |group| group << nil while group.size < number_of_rows }
+                .transpose
+  end
+
+  def format_file_details
+    details = files.map { |file| build_detail(file) }
+    max_length_list = build_max_length_list(details)
+    formatted_details = details.map do |detail|
+      DETAIL_KEYS.map { |key| align_detail(key:, value: detail[key], max_length: max_length_list[key]) }
```

Comment:
```suggestion
      DETAIL_KEYS.map { |key| align_detail(key, detail[key], max_length_list[key]) }
```

キーワード引数にしない方がAPIがすっきりしそうです。

```suggestion
      DETAIL_KEYS.map { |key| align_detail(key, detail, max_length_list) }
```


こうすればさらにスッキリ？

Reply:
キーワード引数を無しにして、ハッシュのまま引数を渡すよう変更しました。https://github.com/SuzukaHori/ruby-practices/pull/12/commits/e3cee449b07fcaa940c19447eeb360a6a8a43bea